api/pom.xml

```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>infinispan-parent</artifactId>
+        <groupId>org.infinispan</groupId>
+        <version>10.0.0-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>infinispan-api</artifactId>
+    <name>Infinispan Api</name>
+    <description>Infinispan Api</description>
+
+    <build>
+        <pluginManagement>
+            <plugins>
+                <plugin>
+                    <artifactId>maven-compiler-plugin</artifactId>
+                    <version>${version.maven-compiler-plugin}</version>
+                    <configuration>
+                        <release>11</release>
```

Comment:
If I'm not wrong here, one of the motivations is to avoid multi-release jars. The dependency to reactive-streams will be added in any case. @tristantarrant any further thing to add here?